solidity/test/isms/CCIPIsm.t.sol

```diff
@@ -0,0 +1,542 @@
+
+        vm.makePersistent(address(ccipISMMainnet), address(ccipHookMainnet), MAINNET_ROUTER_ADDRESS);
+    }
+
+    function deployCCIPHookAndISMOptimism() public {
+        vm.selectFork(optimismFork);
+
+        l2Mailbox = new TestMailbox(MAINNET_DOMAIN);
+
+        ccipISMOptimism = new CCIPIsm();
+
+        ccipHookOptimism = new CCIPHook(
+            OP_ROUTER_ADDRESS,
+            address(l2Mailbox),
+            MAINNET_DOMAIN,
+            TypeCasts.addressToBytes32(address(ccipISMOptimism))
+        );
+
+        ccipHookOptimism.setIsm(address(ccipISMOptimism));
+        ccipHookOptimism.addDestinationChainToAllowlist(MAINNET_CHAIN_SELECTOR, true);
+        ccipHookOptimism.addSourceChainToAllowlist(MAINNET_CHAIN_SELECTOR, true);
+
+        ccipISMOptimism.setAuthorizedHook(TypeCasts.addressToBytes32(address(ccipHookOptimism)));
+
+        vm.makePersistent(address(ccipISMOptimism), address(ccipHookOptimism), OP_ROUTER_ADDRESS);
+    }
+
+    function deployAll() public {
+        deployCCIPHookAndISMMainnet();
+        deployCCIPHookAndISMOptimism();
+
+        vm.selectFork(mainnetFork);
+        ccipHookMainnet.addSenderToAllowlist(address(ccipHookOptimism), true);
+
+        vm.selectFork(optimismFork);
+        ccipHookOptimism.addSenderToAllowlist(address(ccipHookMainnet), true);
+
+        // for sending value
+        vm.deal(
+            OP_ROUTER_ADDRESS,
+            2 ** 255
+        );
+    }
+
+    ///////////////////////////////////////////////////////////////////
+    ///                         FORK TESTS                          ///
+    ///////////////////////////////////////////////////////////////////
+
+    /* ============ hook.quoteDispatch ============ */
+
+    function testFork_quoteDispatch() public {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        testMetadata = _buildMetadata(OPTIMISM_CHAIN_SELECTOR, address(ccipHookOptimism));
+
+        assertEq(ccipHookMainnet.quoteDispatch(testMetadata, encodedMessage), 8891396275644);
+    }
+
+    /* ============ hook.postDispatch ============ */
+
+    function testFork_postDispatch() public {
+        deployAll();
+
+        vm.selectFork(mainnetFork);
+
+        bytes memory encodedHookData = abi.encodeCall(
+            AbstractMessageIdAuthorizedIsm.verifyMessageId,
+            (messageId)
+        );
+
+        _allowAnyAddressToSendCCIP();
+
+        l1Mailbox.updateLatestDispatchedId(messageId);
+
+        vm.expectEmit(false, true, false, false, address(ccipHookMainnet));
+        emit MessageSent(
+            bytes32(0),
+            OPTIMISM_CHAIN_SELECTOR,
+            address(ccipHookOptimism),
+            bytes(encodedHookData),
+            address(0),
+            uint256(8891396275644) 
+        );
+        testMetadata = _buildMetadata(OPTIMISM_CHAIN_SELECTOR, address(ccipHookOptimism));
+
+        ccipHookMainnet.postDispatch{value: 1 ether}(testMetadata, encodedMessage);
```

Comment:
if you test with overpaying, check for refund back in atleast one of the tests

Reply:
No refunds in CCIP, so we can send either exact amount or with some buffer (I set it to 10% currently)